pkg/reconciler/service/runner.go

```diff
@@ -80,22 +81,46 @@ func (r *runner) reconcile(ctx context.Context, model *reconciler.Reconciliation
 		return err
 	}
 
-	chartProvider, err := r.newChartProvider()
+	inClusterClient, err := kubeclient.NewInClusterClient()
+	if err != nil {
+		return err
+	}
+
+	inClusterClientSet, err := inClusterClient.GetClientSet()
+	if err != nil {
+		return err
+	}
+
+	clusterClientSet, err := kubeClient.Clientset()
+	if err != nil {
+		return err
+	}
+
+	configs := model.ConfigsToMap()
+	err = model.Repository.ReadToken(inClusterClientSet.CoreV1(), configs["repo.token.namespace"])
+	if err != nil {
+		return err
+	}
+
+	chartProvider, err := r.newChartProvider(&model.Repository)
 	if err != nil {
 		return errors.Wrap(err, "Failed to create chart provider instance")
 	}
 
-	wsFactory, err := r.workspaceFactory()
+	wsFactory, err := r.workspaceFactory(&model.Repository)
 	if err != nil {
 		return err
 	}
 
 	actionHelper := &ActionContext{
-		KubeClient:       kubeClient,
-		WorkspaceFactory: wsFactory,
-		Context:          ctx,
-		Logger:           r.logger,
-		ChartProvider:    chartProvider,
+		InClusterClientSet: inClusterClientSet,
+		KubeClient:         kubeClient,
+		ClientSet:          clusterClientSet,
+		WorkspaceFactory:   wsFactory,
+		Context:            ctx,
+		Logger:             r.logger,
+		ChartProvider:      chartProvider,
+		ConfigsMap:         configs,
```

Comment:
This should not go into the `ActionContext` as it's purpose is to give access to non-component specific functionality (cross-cutting features).

I would suggest to move the `ConfigToMap()` function from the model into the connectivity-proxy (currently only used there) OR  we pass this map directly to the Actions (I can image this makes life in general easier for our colleagues).

     type Action interface {
	Run(version, profile string, configuration map[string]interface{}, helper *ActionContext) error
     }

Reply:
To be handled on separate branch.

Reply:
Refactored.